plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/util/ResourceUtils.java

```diff
@@ -681,11 +681,11 @@ public static boolean areRelated(final IBuildConfig config, final IDeploymentCon
 	 * @param dc the deployment config to match
 	 * @return true if they are related
 	 */
-    public static boolean areRelated(final IService service, IDeploymentConfig dc) {
+    public static boolean areRelated(final IService service, final IDeploymentConfig dc) {
         return service.getProject().getResources(ResourceKind.POD).stream()
-                                                                   .filter(pod -> containsAll(service.getSelector(), pod.getLabels()))   
-                                                                   .filter(pod -> dc.getName().equals(pod.getAnnotation(OpenShiftAPIAnnotations.DEPLOYMENT_CONFIG_NAME)))   
-                                                                   .count() > 0; 
+        	.filter(pod -> containsAll(service.getSelector(), pod.getLabels()))   
+            .filter(pod -> dc.getName().equals(pod.getAnnotation(OpenShiftAPIAnnotations.DEPLOYMENT_CONFIG_NAME)))   
+            .count() > 0; 
```

Comment:
nitpicking
I think that the first filter is longer running than the second - so it's better to rearrange them for performance, so that currently first filter would work on less number of pods.